sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -429,5 +534,124 @@ protected EventProcessor()
 
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        ///   A virtual <see cref="StorageManager" /> instance that delegates calls to the
+        ///   <see cref="EventProcessor{TPartition}" /> to which it is associated.
+        /// </summary>
+        ///
+        private class DelegatingStorageManager : StorageManager
+        {
+            /// <summary>
+            ///   The <see cref="EventProcessor{TPartition}" /> that the storage manager is associated with.
+            /// </summary>
+            ///
+            private WeakReference<EventProcessor<TPartition>> Processor { get; }
```

Comment:
Interesting - why did you choose to adopt `WeakReference<T>` here?  What would the problem be if this had a strong reference back to the `EventProcessor<TPartition>` in question?

Reply:
You'd have a circular reference between the PartitionManager (via the storage manager) and the Processor.   Granted, they're part of a closed graph and should fall out of reach at the same time, so it's unlikely that we'd see them drift, so unlikely a problem under normal circumstances.

Outside of the bit of added complexity, am I overlooking a scenario where the weak reference is potentially problematic?

Reply:
Thanks for the sanity checks; I appreciate the shin kicking.  Removal to be included in the next PR.